src/main/java/com/movierecommender/project/MovieController.java

```diff
@@ -1,70 +1,68 @@
 package com.movierecommender.project;
+
 import com.movierecommender.project.dto.Movie;
 import com.movierecommender.project.service.IMovieService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.*;
+
 import java.io.BufferedWriter;
 import java.io.FileWriter;
 
-
 @Controller
 public class MovieController {
+
     @Autowired
-    IMovieService movieService;
+    private IMovieService movieService;
 
     // Fetch a specific movie from the external API
     @GetMapping("/movie")
     @ResponseBody
-    public ResponseEntity<?> getMovieFromApi() {
+    public ResponseEntity<?> getMovieFromApi(@RequestParam int movieId) {
         try {
-            Movie movie = movieService.fetchMovieFromExternalApi(24); // Example movie ID
+            Movie movie = movieService.fetchMovieFromExternalApi(movieId);
             return ResponseEntity.ok(movie);
         } catch (Exception e) {
-            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
-                    .body("Error occurred while fetching movie data: " + e.getMessage());
+            return ResponseEntity.internalServerError()
+                    .body("Error while fetching movie: " + e.getMessage());
```

Comment:
I did changes in  ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR) was used explicitly to set the status.
Now, ResponseEntity.internalServerError() is used, which is a shorthand method provided by ResponseEntity. Source: https://www.baeldung.com/spring-response-entity